lifecycle-operator/controllers/lifecycle/schedulinggates/controller_test.go

```diff
@@ -335,3 +335,50 @@ func TestSchedulingGatesReconciler_Reconcile(t *testing.T) {
 		})
 	}
 }
+
+func TestHasKeptnSchedulingGate(t *testing.T) {
+	tests := []struct {
+		name    string
+		pod     *v1.Pod
+		hasGate bool
+	}{
+		{
+			name: "PodWithKeptnSchedulingGate",
+			pod: &v1.Pod{
+				Spec: v1.PodSpec{
+					SchedulingGates: []v1.PodSchedulingGate{
+						{
+							Name: apicommon.KeptnGate,
+						},
+					},
+				},
+			},
+			hasGate: true,
+		},
+		{
+			name:    "PodWithoutSchedulingGate",
+			pod:     &v1.Pod{},
+			hasGate: false,
+		},
+		{
+			name: "PodWithOtherSchedulingGates",
+			pod: &v1.Pod{
+				Spec: v1.PodSpec{
+					SchedulingGates: []v1.PodSchedulingGate{
+						{
+							Name: "other-gate",
+						},
+					},
+				},
+			},
+			hasGate: false,
+		},
+	}
```

Comment:
let's add a test with 2 gates: one Keptn and then some other, this use-case we are missing

Reply:
I have added a test with two gates